server/sandbox_network.go

```diff
@@ -55,10 +55,12 @@ func (s *Server) networkStart(ctx context.Context, sb *sandbox.Sandbox) (podIPs
 		return
 	}
 
-	for _, podIPConfig := range network.IPs {
+	for idx, podIPConfig := range network.IPs {
 		podIP := strings.Split(podIPConfig.Address.String(), "/")[0]
 
-		if len(sb.PortMappings()) > 0 {
+		// Apply the hostport mappings only for the first IP to avoid allocating
+		// the same host port twice
+		if idx == 0 && len(sb.PortMappings()) > 0 {
```

Comment:
If idx[0]'s Portmappings is NOT > zero, should we keep checking until we find one that is?

Reply:
Hm, no this should be fine since the port mappings are sandbox 'global' and independently specified from the `network.IPs`. :innocent: